### PR TITLE
[INFRA-4975] Replaced existing Jenkinsfile with one suitable for us

### PR DESCRIPTION
**Pull request by ajonnavi**
## Why
To enable builds from our Jenkins nodes.

## How
1. Change version number to reflect 6si builds
2. Add workflows for dev and prod builds

## Testing evidence
```
aws s3 sync --exclude '*' --include '*-bin.tar.gz' ./packaging/target $BOOTSTRAP_BUCKET/hive/dev/ --dryrun
(dryrun) upload: packaging/target/apache-hive-2.3.9-6si-1-bin.tar.gz to s3://<our bootstrap bucket>/hive/dev/apache-hive-2.3.9-6si-1-bin.tar.gz
```

### CODE REVIEW

File: jdbc-handler/pom.xml [hunk 1]
```diff
@@ -29,6 +29,7 @@
 
   <properties>
     <hive.path.to.root>..</hive.path.to.root>
+    <commons-dbcp.version>1.4</commons-dbcp.version>
```
Comment (davidecerri) [inline jdbc-handler/pom.xml:32]: what is this?

Comment (davidecerri) [reply jdbc-handler/pom.xml:32]: just google it: https://commons.apache.org/proper/commons-dbcp/

**Review by davidecerri — APPROVED** [review]

File: jdbc-handler/pom.xml (hunk 1 above)
Comment (ajonnavi) [reply jdbc-handler/pom.xml:32]: Yeah it's strange that it's missing. When I try to build it on my laptop, it was complaining that this dependency was missing. They even have the version listed in the top level `pom.xml` - https://github.com/6si/hive/blob/6si-main/pom.xml#L132, but this is the module that actually needs it and it's not listed there.

**Comment by shakti-6s** [comment]
@avadhanij6si This PR have tests skipped in the jenkins build flow. Will there be a following PR with the fix to run tests for prod build flow?

**Comment by ajonnavi** [comment]
@shakti-6s, we will need to discuss this in the next meeting when Ashish is back, but creating the environment for running the full suite of tests is a non trivial task.

We need to figure out how much of it is practical, and how much is possible given our bandwidth. We will need to rely on our own testing in the meantime.

**Review by ashishkumar1910 — APPROVED** [review]

**Pull request merged**